### DISCUSSION TITLE
Perf Query fix feedback

### DISCUSSION THREAD

**Discussion by Medard22** [Medard22/Dolphin-MMJR2-VBI#34] [Bug Report]
I need help with testing one particular PR. There is an update on Dolphin Official that fixes Queries in the Vulkan renderer. Perf queries are used heavily in Super Mario Sunshine. However, on my SD855+ device, in Super Mario Sunshine the change causes graphical glitches, framerate drops, and app freeze unless you turn on "Skip EFB Access from CPU". I think it might be an Adreno Vulkan driver bug specific to my device. This build has no issues on Mali devices (tested on RP3+ and Samsung S9+). I tried the latest Dolphin Official and it glitched and froze as well. I want to test this build on more devices, preferably with Adreno GPU before I merge it.

**Comment by EdwardTheBanker** [comment]
Hi bro i have Dimensity 1200 and snap 8 gen 1. What do you need exactly?

**Reply by Medard22** [reply]
Please update to the latest dev version and then try running Super Mario Sunshine with "Skip EFB Access from CPU" set to off. It should work  ok, but on my SD855+ it freezes before you can pick save file slot.  Thank you.

**Comment by EdwardTheBanker** [comment]
Both dimensity 1200 and Snap 8 Gen 1 works fine with Skip EFB Access CPU on and Off at vulkan and OpenGL.   
  

  

  

  
  
  
  
>

**Reply by Medard22** [reply]
Thank you for your feedback 👍

**Comment by panjidharma** [comment]
Bro update GPU driver please

**Reply by Medard22** [reply]
Sorry I am not sure if I follow. My device is already EOL, there are no further updates planned. As for other devices it´s usually up to manufacturers to keep drivers up-to-date. For this particular change I am just collecting feedback.

**Reply by panjidharma** [reply]
No, i mean dolphin official add custom turnip Driver, can you implemented in mmjr?

**Reply by Medard22** [reply]
That would require integrating Adreno Tools into Dolphin. I am afraid that is way beyond what I can do. I could port it over if someone from Dolphin Official implements it. But I really can´t implement it from scratch.

**Comment by Medard22** [comment]
Thanks to everyone who helped with the testing. The Perf Query fix has been merged to the latest release candidate branch.

**Comment by Kirbythepink1980** [comment]
Bro can u add the "Emulated USB devices" so we can play more games that need portal like skylanders. The "emulated usb devices" are on the latest development version. Config - wii - emulated usb devices. Thank you in advance if you make it and Thank you again if you are no interested.

**Reply by Kirbythepink1980** [reply]
*Not

**Reply by Medard22** [reply]
Let´s continue our discussion here and not piggyback on a closed topic - https://github.com/Medard22/Dolphin-MMJR2-VBI/discussions/63
